09 Redux/src/components/member/memberPage.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { connect } from 'react-redux'
+import { hashHistory } from 'react-router'
+import * as toastr from 'toastr';
+import MemberEntity from './../../api/memberEntity'
+import MemberForm from './memberForm';
+import MemberAPI from '../../api/memberAPI';
+import objectAssign = require('object-assign');
+import loadMember from '../../actions/loadMember'
+import saveMember from '../../actions/saveMember'
+import MemberErrors from  '../../validations/MemberFormErrors'
+import uiInputMember from '../../actions/uiInputMember'
+import resetSaveCompleted from '../../actions/resetSaveCompleted'
+import newMember from  '../../actions/newMember'
+
+interface Props extends React.Props<MemberPage> {
+  params : any
+  member? : MemberEntity
+  ,errors?: MemberErrors
+  ,saveCompleted? : boolean
+  ,loadMember? : (id : number) => void
+  ,fireValidationFieldValueChanged  : (fieldName : string, value : any) => void
+  ,saveMember: () => void
+  ,newMember: () => void
+  ,resetSaveCompletedFlag: () => void
+}
+
+class MemberPage extends React.Component<Props, {}> {
+
+  constructor(props : Props){
+        super(props);
+
+  }
+
+  componentWillMount() {
+    // Coming from navigation
+    var memberId = this.props.params.id;
+
+    if(memberId) {
+      var memberIdNumber : number = parseInt(memberId);
+      this.props.loadMember(memberIdNumber);
+    } else {
+      this.props.newMember();
+    }
+  }
+
+ // https://github.com/reactjs/redux/issues/580
+ componentWillReceiveProps(nextProps) {
+   if(this.props.saveCompleted != nextProps.saveCompleted
+      && nextProps.saveCompleted == true) {
```

Comment:
Does not needed '== true'

`if (this.props.saveCompleted != nextProps.saveCompleted && nextProps.saveCompleted)`